tests/unit/client_test.py

```
@@ -249,3 +252,289 @@ def test_set_default_max_query_payment_non_finite_value(invalid_amount):
+        client.generate_transaction_id()
+
+    client.close()
+
+
+def test_generate_transaction_id_returns_transaction_id(monkeypatch):
+    """Test that generate_transaction_id returns a TransactionId object when operator is set."""
+    client = Client.for_testnet()
+    client.operator_account_id = AccountId(0, 0, 1234)
+
+    fake_txid = MagicMock(spec=TransactionId)
+    monkeypatch.setattr("hiero_sdk_python.transaction.transaction_id.TransactionId.generate", lambda account_id: fake_txid)
```

Ruff (0.14.14)

[warning] Unused lambda argument: `account_id`

(ARG005)